src/main/java/org/terasology/ligthandshadow/componentsystem/controllers/ScoreSystem.java

```diff
@@ -0,0 +1,103 @@
+/*
+ * Copyright 2017 MovingBlocks
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.terasology.ligthandshadow.componentsystem.controllers;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.terasology.entitySystem.entity.EntityManager;
+import org.terasology.entitySystem.entity.EntityRef;
+import org.terasology.entitySystem.event.ReceiveEvent;
+import org.terasology.entitySystem.systems.BaseComponentSystem;
+import org.terasology.entitySystem.systems.RegisterMode;
+import org.terasology.entitySystem.systems.RegisterSystem;
+import org.terasology.ligthandshadow.componentsystem.components.LASTeamComponent;
+import org.terasology.ligthandshadow.componentsystem.components.WinConditionCheckOnActivateComponent;
+import org.terasology.logic.characters.CharacterHeldItemComponent;
+import org.terasology.logic.common.ActivateEvent;
+import org.terasology.logic.inventory.InventoryManager;
+import org.terasology.registry.In;
+import org.terasology.rendering.nui.ControlWidget;
+import org.terasology.rendering.nui.NUIManager;
+import org.terasology.rendering.nui.databinding.ReadOnlyBinding;
+import org.terasology.rendering.nui.widgets.UILabel;
+import org.terasology.world.block.items.BlockItemComponent;
+
+@RegisterSystem(RegisterMode.AUTHORITY)
+public class ScoreSystem extends BaseComponentSystem {
+    private static final Logger logger = LoggerFactory.getLogger(ScoreSystem.class);
+    private static final String BLACK_FLAG_URI = "LightAndShadowResources:blackFlag";
+    private static final String RED_FLAG_URI = "LightAndShadowResources:redFlag";
```

Comment:
You are defining the same constants in multiple locations (ScoreSystem, AttackSystem). Ideally, it should be statically available from done utility context.

Reply:
I know this is being done wrong at the moment (constants should only be defined in one place) but wasn't sure where that one place should be. What do you mean by "done utility context"? I was thinking of defining them in the `FlagComponent` file and accessing from there, would that make sense?

Reply:
Should have read "some utility". Not sure whether components should have constants specific to instances kind of but it's a start

Reply:
Nah, shouldn't be in the component. @dacharya64 what @skaldarnar meant was to have a Utility class outside that's probably package private and holds the strings as `public static final` making them accessible to multiple other classes from the same package.